embrace-android-sdk/src/test/java/io/embrace/android/embracesdk/internal/compression/GzipCompressorTest.kt

```diff
@@ -0,0 +1,30 @@
+package io.embrace.android.embracesdk.internal.compression
+
+import io.embrace.android.embracesdk.logging.InternalEmbraceLogger
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import java.io.ByteArrayOutputStream
+import java.util.zip.GZIPInputStream
+
+internal class GzipCompressorTest {
+
+    private lateinit var compressor: GzipCompressor
+
+    @Before
+    fun setUp() {
+        compressor = GzipCompressor(logger = InternalEmbraceLogger())
+    }
+
+    @Test
+    fun testCompression() {
```

Comment:
Would be good to have a test case for when the payload is not compressed too

Reply:
Here the compressor class just compresses what appears in the outputStream of the SerializationAction provided. The tests for covering the behavior of when the stored payload is compressed or not are in these tests:
- `test cachePayload stores uncompressed data and loadPayload returns compressed data`
- `test cachePayload stores compressed data and loadPayload returns compressed data`
in the `EmbraceCacheServiceTest` class.